server/src/test/java/io/crate/execution/engine/collect/files/FileReadingCollectorTest.java

```diff
@@ -71,14 +74,24 @@
 import io.crate.types.DataTypes;
 
 public class FileReadingCollectorTest extends ESTestCase {
-
+    private static ThreadPool THREAD_POOL;
     private static File tmpFile;
     private static File tmpFileGz;
     private static File tmpFileEmptyLine;
     private InputFactory inputFactory;
     private Input<String> sourceUriFailureInput;
     private TransactionContext txnCtx = CoordinatorTxnCtx.systemTransactionContext();
 
+    @BeforeClass
+    public static void setupThreadPool() {
+        THREAD_POOL = new TestThreadPool(Thread.currentThread().getName());
+    }
```

Comment:
Isn't it flaky to rely on that this is called before `setUpClass`? why not do this inside `setUpClass`?

Reply:
Thank you, done